src/Features/CSharpTest/ImplementAbstractClass/ImplementAbstractClassTests_ThroughMember.cs

```diff
@@ -826,5 +826,107 @@ public override void Method1()
                 </Workspace>
                 """, index: 1);
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69177")]
+        public async Task TestImplementThroughPrimaryConstructorParam1()
+        {
+            await TestInRegularAndScriptAsync(
+                """
+                abstract class Base
+                {
+                    public abstract int Method();
+                }
+
+                class [|Program|](Base base1) : Base
+                {
+                }
+                """,
+                """
+                abstract class Base
+                {
+                    public abstract int Method();
+                }
+
+                class Program(Base base1) : Base
+                {
+                    public override int Method()
+                    {
+                        return base1.Method();
+                    }
+                }
+                """, index: 1);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69177")]
+        public async Task TestImplementThroughPrimaryConstructorParam2()
+        {
+            // Don't offer "implement through" since this PC parameter is captured as a field.
+            await TestExactActionSetOfferedAsync(
+                """
+                abstract class Base
+                {
+                    public abstract int Method();
+                }
+
+                class [|Program|](Base base1) : Base
+                {
+                    private Base _base = base1;
+                }
+                """, new[] { FeaturesResources.Implement_abstract_class, string.Format(FeaturesResources.Implement_through_0, "_base") });
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/69177")]
+        public async Task TestImplementThroughPrimaryConstructorParam2_B()
+        {
+            // Don't offer "implement through 'base1'" since this PC parameter is captured as a field.
+            await TestExactActionSetOfferedAsync(
+                """
+                abstract class Base
+                {
+                    public abstract int Method();
+                }
+
+                class [|Program|](Base base1) : Base
+                {
+                    private Base _base = (base1);
```

Comment:
💡 Consider adding a case where it's used in a field initializer, but not for keeping a reference. For example:

```cs
primary int _base1HashCode = base1.GetHashCode();
```